Flaky test: TestTDHttpClient#retryOn429

See https://app.circleci.com/insights/github/treasure-data/td-client-java/workflows/workflow/tests?branch=master
See https://app.circleci.com/pipelines/github/treasure-data/td-client-java/270/workflows/51382faf-0cd2-414c-953e-a6d8b21d2e40/jobs/1088

`com.treasuredata.client.TestTDHttpClient#retryOn429` fails quite often and retry fixes it.
The test case asserts that system waits for the given timeout (5 sec).
It looks like that timing is not so accurate in micro-seconds order.
```
java.lang.AssertionError: 
Expected: a value equal to or greater than <5000000000L>
     but: <4999688441L> was less than <5000000000L>
	at com.treasuredata.client.TestTDHttpClient.retryOn429(TestTDHttpClient.java:176)
```

--- a/src/test/java/com/treasuredata/client/TestTDHttpClient.java
+++ b/src/test/java/com/treasuredata/client/TestTDHttpClient.java
@@ -46,6 +46,7 @@ import java.util.concurrent.atomic.AtomicLong;
 
 import static com.google.common.net.HttpHeaders.USER_AGENT;
 import static com.treasuredata.client.TDHttpRequestHandlers.stringContentHandler;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.exparity.hamcrest.date.DateMatchers.within;
 import static org.hamcrest.Matchers.is;
@@ -172,8 +173,11 @@ public class TestTDHttpClient
         assertThat(requests.get(), is(2));
         assertThat(result, is(body));
 
+        // The delay often goes below 5 seconds fractionally, e.g. 4.9996 seconds (99.992%),
+        // which appears to be acceptable. So 1% margin of error is allowed.
         long delayNanos = secondRequestNanos.get() - firstRequestNanos.get();
-        assertThat(delayNanos, Matchers.greaterThanOrEqualTo(SECONDS.toNanos(retryAfterSeconds)));
+        long almostFiveSeconds = MILLISECONDS.toNanos(retryAfterSeconds * 990);
+        assertThat(delayNanos, Matchers.greaterThanOrEqualTo(almostFiveSeconds));
     }
 
     @Test

